app/reducers/views/channel.js

```diff
@@ -18,156 +18,199 @@ function displayName(state = '', action) {
     }
 }
 
-function drafts(state = {}, action) {
-    switch (action.type) {
-    case ViewTypes.POST_DRAFT_CHANGED: {
-        return {
-            ...state,
-            [action.channelId]: Object.assign({}, state[action.channelId], {draft: action.postDraft})
-        };
-    }
-    case ViewTypes.SET_POST_DRAFT: {
-        return {
+function handlePostDraftChanged(state, action) {
+    return {
+        ...state,
+        [action.channelId]: Object.assign({}, state[action.channelId], {draft: action.draft})
+    };
+}
+
+function handlePostDraftSelectionChanged(state, action) {
+    return {
+        ...state,
+        [action.channelId]: Object.assign({}, state[action.channelId], {
+            cursorPosition: action.cursorPosition
+        })
+    };
+}
+
+function handleSetPostDraft(state, action) {
+    return {
+        ...state,
+        [action.channelId]: {
+            draft: action.draft,
+            cursorPosition: 0,
```

Comment:
Should this mangle the cursorPosition? Or is the selection changed event always fired after this one?

Reply:
It's the other way around, the selection change event occurs first followed by the onChanged, but from my tests the cursor position is correct